x/posts/internal/types/post_test.go

```diff
@@ -23,6 +23,38 @@ func TestPostID_UnmarshalJSON(t *testing.T) {
 	assert.Equal(t, types.PostID(10), id)
 }
 
+func TestPostIDs_AppendIfMissing(t *testing.T) {
+	tests := []struct {
+		name     string
+		id       types.PostID
+		expLen   int
+		expFound bool
+	}{
+		{
+			name:     "AppendIfMissing dont append anything",
+			id:       types.PostID(1),
+			expLen:   0,
+			expFound: false,
+		},
+		{
+			name:     "AppendIfMissing append something",
+			id:       types.PostID(2),
+			expLen:   2,
+			expFound: true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			ids := types.PostIDs{types.PostID(1)}
```

Comment:
I would avoid doing such initialization as it makes the whole test less clear. What I would do, instead, is adding a new field to the `tests` structure to initialize the IDs array. I would also change `expLen` with a new list of PostID to match for it:

```golang
tests := []struct {
  name      string
  IDs       types.PostIDs
  newID     types.PostID
  expIDs    types.PostIDs
  expEdited bool
} { ... }
```


```golang
newIDs, edited := test.IDs.AppendIfMissing(test.ID) 
assert.Equal(t, test.expIDs, newIDs)
assert.Equal(t, test.expEdited, edited)
```

Reply:
Sure, and sorry for this, I'm new to this way of testing, I'm trying to get used to it as fast as I can!